src/packagedcode/models.py

```diff
@@ -94,14 +103,80 @@
   -- code in source or compiled form or both.
 """
 
-class Package(object):
+
+class EnhancedModel(Model):
+    def __init__(self, deserialize_mapping=None, strict=True, **kwargs):
+        Model.__init__(self, raw_data=kwargs, deserialize_mapping=deserialize_mapping, strict=strict)
+
+
+class Version(EnhancedModel):
```

Comment:
I am a tad confused about Version being a model vs being what used to be called versioning? can you explain how you see this work in practice?